bin/replace

```diff
@@ -9,4 +9,4 @@ shift
 replace_with="$1"
 shift
 
-ag -l --nocolor "$find_this" $* | xargs sed -i '' "s/$find_this/$replace_with/g"
+ag -l --nocolor "$find_this" "$@" | xargs sed -i '' "s/$find_this/$replace_with/g"
```

Comment:
There are two things happening here, `$*` vs `$@` and quoting.

Here's a shell script that will illustrate the difference:

```sh
#!/bin/sh

for argument in $*; do
  echo "$argument"
done
```

(For each bit about `$*` vs `$@` vs `"$@"`, assume I changed the `for` loop in the script accordingly.)

`$*`, with no quotes, will totally ignore spaces in arguments, even if the arguments are quoted:

```sh
# $*
$ ./demo.sh "keep it together" two
keep
it
together
too
```

Now let's try it with `"$*"`:

```sh
# "$*"
$ ./demo.sh "keep it together" two
keep it together too
```

As you can see, it interpreted all arguments as one huge string, instead of `keep it together` and `two` as separate strings. (So, it's wrong.)

Now let's try with `$@`:

```sh
# $@
$ ./demo.sh "keep it together" two
keep
it
together
too
```

It acts just like `$*`.

And finally, the one that we almost always want to use, `"$@"`:

```sh
# "$@"
$ ./demo.sh "keep it together" two
keep it together
too
```

At last, we have printed two lines for two arguments, appropriately split up according to how we passed them in as arguments.